src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,92 @@
+/* eslint-disable no-console */
 import { useState } from 'react';
+import { Movie } from '../../types/Movie';
 import { TextField } from '../TextField';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+interface Props {
+  onAdd: (movie: Movie) => void;
+}
+
+export const NewMovie = (props: Props) => {
+  const { onAdd } = props;
+
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const isButtonAvailable = () => {
+    if (title && imdbId && imgUrl && imdbUrl) {
+      return true;
+    }
+
+    return false;
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={(event) => {
+        event.preventDefault();
+
+        setCount(prev => prev + 1);
+        setTitle('');
+        setDescription('');
+        setImgUrl('');
+        setImdbUrl('');
+        setImdbId('');
+
+        onAdd({
+          title,
+          description,
+          imgUrl,
+          imdbUrl,
+          imdbId,
+        });
+      }}
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={title}
+        onChange={(eventValue) => setTitle(eventValue)}
```

Comment:
Try, should work the same

```suggestion
        onChange={setTitle}
```